x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotIndexInput.java

```diff
@@ -259,25 +257,8 @@ public String toString() {
     }
 
     private InputStream openBlobStream(int part, long pos, long length) throws IOException {
-        final InputStream stream;
-        if (fileInfo.metadata().hashEqualsContents() == false) {
-            stream = blobContainer.readBlob(fileInfo.partName(part), pos, length);
-        } else {
-            // extract blob content from metadata hash
-            final BytesRef data = fileInfo.metadata().hash();
-            if (part > 0) {
-                assert fileInfo.numberOfParts() >= part;
-                for (int i = 0; i < part; i++) {
-                    pos += fileInfo.partBytes(i);
-                }
-            }
-            if ((pos < 0L) || (length < 0L) || (pos + length > data.bytes.length)) {
-                throw new IllegalArgumentException("Invalid arguments (pos=" + pos + ", length=" + length
-                    + ") for hash content (length=" + data.bytes.length + ')');
-            }
-            stream = new ByteArrayInputStream(data.bytes, Math.toIntExact(pos), Math.toIntExact(length));
-        }
-        return stream;
+        assert assertHashIsNotEqualToContent();
```

Comment:
If we move the assertion to the constructor then this isn't necessary.

Reply:
+1